typed-racket-doc/typed-racket/scribblings/reference/types.scrbl

```diff
@@ -974,6 +974,31 @@ prefab types with the (implicitly quoted) prefab-key
 
 @history[#:added "1.7"]}
 
+@defform[(Struct-Property ty)]{
+  Internal type that describes a property value that is attached to a
+   structure type.
+
+  @ex[(:print-type prop:input-port)]
+
+  @history[#:added "1.10"]}
+
+@defidform[Self]{
+  Internal type that can only appear in a @racket[Struct-Property] type.
+  Refers to an instance of a structure type that has a value for the property.
```

Comment:
Refers to "an" instance vs refers to "the" instance? I thought `Self` was only meant for "the" exact instance that triggered the property lookup, while `Imp` was for "an" instance of the same structure type.